File: Solutions in Javascript/FactorialTrailingZeroes.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {number} n
 * @return {number}
 */
var trailingZeroes = function(n) {
    var ans = 0;
    var fivePower = 5;
    while(Math.floor(n / fivePower) > 0){
        ans += Math.floor(n / fivePower); 
        fivePower *= 5;
    }
    return ans;
};

File: Solutions in Java/FactorialTrailingZeroes.java
/**
 * @author : Anmol Agrawal
 */
/* Logic 1: */
/*
class Solution {
    public int trailingZeroes(int n) {
        int ans = 0;
        while(n > 0){
            n /= 5;
            ans += n;
        }
        return ans;
    }
}
*/
/* Logic 2: Recursion */
class Solution {
    public int trailingZeroes(int n) {
        if(n == 0) return 0;
        return (n / 5) + trailingZeroes(n / 5);
    }
}